pkg/client/cache/reflector.go

```diff
@@ -161,7 +161,8 @@ func (r *Reflector) listAndWatch(stopCh <-chan struct{}) {
 			default:
 				glog.Errorf("Failed to watch %v: %v", r.expectedType, err)
 			}
-			return
+			time.Sleep(time.Second)
```

Comment:
Ah, you're right, I got confused about which error handler I was looking at. I think the comments in the error cases here are totally wrong. The watch can't "close normally" when the error is a failure to start the watch. And how is there a difference between EOF and UnexpectedEOF here?

I am still pretty worried that there's a class of error that will cause this loop to never terminate. I think if you want to retry here you need to have a counter so that it gives up and starts again with a list after e.g. 3 tries or something.